src/eureka/lib/util.py

```diff
@@ -92,6 +92,56 @@ def trim(data, meta):
     return subdata, meta
 
 
+def manual_clip(spec, lc, meta, log):
```

Comment:
When moved to S5, only `lc` should be modified, thus `spec` won't be needed as an input parameter.